src/util/entity.cpp

```diff
@@ -193,43 +193,33 @@ namespace big::entity
 
 	bool load_ground_at_3dcoord(Vector3& location)
 	{
-		float groundZ;
-		bool done = false;
+	    constexpr float max_ground_check = 1000.f;
+	    constexpr int max_attempts = 300;
+	    float ground_z = location.z;
+	    int current_attempts = 0;
+	    bool found_ground;
 
-		for (int i = 0; i < 10; i++)
-		{
-			for (int z = 0; z < 1000; z += 25)
-			{
-				float ground_iteration = static_cast<float>(z);
-				// Only request a collision after the first try failed because the location might already be loaded on first attempt.
-				if (i >= 1 && (z % 100 == 0))
-				{
-					STREAMING::REQUEST_COLLISION_AT_COORD(location.x, location.y, ground_iteration);
-					script::get_current()->yield();
-				}
+	    do {
+	        found_ground = MISC::GET_GROUND_Z_FOR_3D_COORD(location.x, location.y, max_ground_check, &ground_z, FALSE, FALSE);
+	        STREAMING::REQUEST_ADDITIONAL_COLLISION_AT_COORD(location.x, location.y, location.z);
 
-				if (MISC::GET_GROUND_Z_FOR_3D_COORD(location.x, location.y, ground_iteration, &groundZ, false, false))
-				{
-					location.z = groundZ + 1.f;
-					done       = true;
-				}
-			}
+	        if (current_attempts % 10 == 0)
+	        {
+	            location.z += 25.f;
```

Comment:
If my math is correct here, you're going to only top off here at 750 height units. Isn't the max height for GTA 800 or something?

Reply:
Chiliad is around the ~800-1000 range, couldn’t say anything about the actual height.

Reply:
When MBG first submitted his new teleport code, we had a user create a ticket about how he couldn't teleport to the top of the Maze Bank tower. IDK how tall it was, but IIRC it was 800 something units, and I had to adjust his code to fix it.

Reply:
Well, i have tested it and it works just fine.
highest ground collision point is: 806
![image](https://github.com/YimMenu/YimMenu/assets/60616540/49eb018d-bd2c-498a-8c41-417cddfbb10c)